### PR TITLE
feat: added inject params and inject query params

### LINKED ISSUE #137: [RFC] Inject route params as signal
We recently talked about the `injectParamsMap` function on 𝕏, It may be a good idea to add them here as well.

Here's the code I'm talking about:

```ts
export function injectParamsMap(injector?: Injector): Signal<ParamMap> {
  return assertInjector(injectParamsMap, injector, () => {
    const route = inject(ActivatedRoute);
    return toSignal(route.paramMap, {
      initialValue: route.snapshot.paramMap,
    });
  });
}
```

It can include three functions, `injectParamsMap`, `injectQueryParams`, and `injectRouteData`. 

Let me know what you think!

### PR DESCRIPTION
**Pull request by eneajaho**
https://twitter.com/Enea_Jahollari/status/1720078134719152200 this tweet but better and with tests 😄

close #137

### CODE REVIEW

**Comment by nartc** [comment]
fix #137

File: libs/ngxtension/inject-params/src/inject-params.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+export function injectParams<T>(
+	keyOrTransform?: any
+): Signal<T | Params | string | null> {
+	const route = inject(ActivatedRoute);
+	const params = route.snapshot.params || {};
+
+	if (typeof keyOrTransform === 'function') {
+		return toSignal(
+			route.params.pipe(startWith(keyOrTransform(params)), map(keyOrTransform)),
+			{ requireSync: true }
```
Comment (JeanMeche) [inline libs/ngxtension/inject-params/src/inject-params.ts:45]: wouldn't the default value make it more reable ? (and also drop that rxjs import)

File: libs/ngxtension/inject-params/src/inject-params.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+ *
+ * @example
+ * const userId = injectParams('id'); // returns the value of the 'id' param
+ * const userId = injectParams(p => p['id'] as string); // same as above but can be used with a custom transform function
+ * const params = injectParams(); // returns the entire params object
+ *
+ * @param keyOrTransform OPTIONAL The key of the param to return, or a transform function to apply to the params object
+ */
+export function injectParams<T>(
+	keyOrTransform?: any
```
Comment (JeanMeche) [inline libs/ngxtension/inject-params/src/inject-params.ts:37]: nit: maybe drop that `any` ?

File: libs/ngxtension/inject-query-params/src/inject-query-params.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+		return toSignal(
+			route.queryParams.pipe(
+				startWith(keyOrTransform(queryParams)),
+				map(keyOrTransform)
+			),
+			{ requireSync: true }
+		);
+	}
+
+	const key = keyOrTransform as string;
```
Comment (JeanMeche) [inline libs/ngxtension/inject-query-params/src/inject-query-params.ts:54]: By replacing any, you wont have to do this type assertion !

File: libs/ngxtension/inject-params/src/inject-params.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+	if (typeof keyOrTransform === 'function') {
+		return toSignal(route.params.pipe(map(keyOrTransform)), {
+			initialValue: keyOrTransform(params),
+		});
+	}
+
+	const getParam = (params: Params) =>
+		keyOrTransform && params && Object.keys(params).length > 0
+			? params[keyOrTransform] ?? null
+			: params;
```
Comment (JeanMeche) [inline libs/ngxtension/inject-params/src/inject-params.ts:51]: Wdyt of this ? 
```suggestion
		keyOrTransform ? params?.[keyOrTransform] ?? null : params;
```

Comment (eneajaho) [reply libs/ngxtension/inject-params/src/inject-params.ts]: Agree

**Review by JeanMeche — APPROVED** [review]

**Review by nartc — APPROVED** [review]

**Pull request merged**